pkg/helpers/klusterletconfig.go

```diff
@@ -0,0 +1,32 @@
+package helpers
+
+import (
+	"fmt"
+
+	listerklusterletconfigv1alpha1 "github.com/stolostron/cluster-lifecycle-api/client/klusterletconfig/listers/klusterletconfig/v1alpha1"
+	klusterletconfighelper "github.com/stolostron/cluster-lifecycle-api/helpers/klusterletconfig"
+	klusterletconfigv1alpha1 "github.com/stolostron/cluster-lifecycle-api/klusterletconfig/v1alpha1"
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func GetMergedKlusterletConfigWithGlobal(
+	klusterletconfigName string,
+	kcLister listerklusterletconfigv1alpha1.KlusterletConfigLister,
+) (*klusterletconfigv1alpha1.KlusterletConfig, error) {
+	var err error
+	var kc *klusterletconfigv1alpha1.KlusterletConfig
+	if klusterletconfigName != "" {
+		kc, err = kcLister.Get(klusterletconfigName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get klusterletconfig %s: %v", klusterletconfigName, err)
+		}
+	}
+
+	globalKlusterletConfig, err := kcLister.Get(constants.GlobalKlusterletConfigName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get global klusterletconfig: %v", err)
+	}
+
+	return klusterletconfighelper.MergeKlusterletConfigs(globalKlusterletConfig, kc)
```

Comment:
do you support the nil input?

Reply:
Yes, this PR https://github.com/stolostron/cluster-lifecycle-api/pull/25 will filter out the nil items.